src/components/ec-phone-number-input/ec-phone-number-input.vue

```diff
@@ -0,0 +1,432 @@
+    >
+      <span>{{ bottomNote }}</span>
+      <ec-icon
+        v-if="isWarning && warningTooltipMessage"
+        v-ec-tooltip="{ content: warningTooltipMessage }"
+        class="ec-phone-number-input__warning-tooltip"
+        data-test="ec-phone-number-input__warning-tooltip"
+        type="warning"
+        name="simple-error"
+        :size="14"
+      />
+    </div>
+
+    <div v-if="hasBottomCTA">
+      <slot name="bottomCTA" />
+    </div>
+  </div>
+</template>
+
+<script>
+import { getUid } from '../../utils/uid';
+import { mask } from '../../utils/mask';
+
+import EcInputField from '../ec-input-field';
+import EcDropdown from '../ec-dropdown';
+import EcIcon from '../ec-icon';
+import EcTooltip from '../../directives/ec-tooltip';
+
+export default {
+  name: 'EcPhoneNumberInput',
+  components: {
+    EcInputField,
+    EcDropdown,
+    EcIcon,
+  },
+  directives: { EcTooltip },
+  model: {
+    prop: 'value',
+    event: 'value-change',
+  },
+  props: {
+    value: {
+      type: Object,
+    },
+    label: {
+      type: String,
+    },
+    note: {
+      type: String,
+    },
+    bottomNote: {
+      type: String,
+    },
+    isWarning: {
+      type: Boolean,
+      default: false,
+    },
+    warningTooltipMessage: {
+      type: String,
+    },
+    errorMessage: {
+      type: String,
+    },
+    errorTooltipMessage: {
+      type: String,
+    },
+    countries: {
+      type: Array,
+    },
+    areCountriesLoading: {
+      type: Boolean,
+      default: false,
+    },
+    isDisabled: {
+      type: Boolean,
+      default: false,
+    },
+    disabledTooltipMessage: {
+      type: String,
+    },
+    isSensitive: {
+      type: Boolean,
+      default: false,
+    },
+    countryPlaceholder: {
+      type: String,
+    },
+    phoneNumberPlaceholder: {
+      type: String,
+    },
+    isSearchEnabled: {
+      type: Boolean,
+      default: true,
+    },
+    searchCountryPlaceholder: {
+      type: String,
+      default: 'Search...',
+    },
+    noCountriesText: {
+      type: String,
+      default: 'No results found',
+    },
+  },
+  data() {
+    return {
+      uid: getUid(),
+      countriesHasFocus: false,
+      previousNumberValue: null,
+      popperModifier: {
+        setPopperWidth: {
+          enabled: true,
+          order: 841,
+          fn: /* istanbul ignore next */ (data) => {
+            data.styles.width = this.$refs.popperWidthReference.offsetWidth;
+            return data;
+          },
+        },
+      },
+      popoverOptions: {
+        placement: 'bottom-end',
+      },
+    };
+  },
+  computed: {
+    id() {
+      return `ec-phone-number-input-field-${this.uid}`;
+    },
+    errorId() {
+      return this.isInvalid ? `ec-phone-number-input-${this.uid}` : null;
+    },
+    countriesItems() {
+      return this.countries.map(country => ({
+        value: country.value,
+        text: country.text + country.value,
```

Comment:
Do we intentionally not have a space between the country name and the area code?

Reply:
With or without I don't think makes much difference since this text is not visible to the user. 
Is only used for searching. I needed somewhere the concatenated text of (country name + area code) so i can search on it.
Otherwise the users were going to search only the name but not the area code.

Reply:
My concern is because the text is actually visible. It is used like tooltip for the items:
![image](https://user-images.githubusercontent.com/4802418/149762072-a9f3aab5-f77d-437a-90d0-410cb49aae7c.png)

Reply:
Amended with space then :)